apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/CourseNumberSearchBar.tsx

```diff
@@ -14,27 +14,52 @@ class CourseNumberSearchBar extends PureComponent<Record<string, never>, CourseN
     }
 
     getCourseNumber() {
-      return RightPaneStore.getUrlCourseNumValue().trim() 
-        ? this.updateCourseNumAndGetFormData() 
-        : RightPaneStore.getFormData().courseNumber;
+        return RightPaneStore.getUrlCourseNumValue().trim()
+            ? this.updateCourseNumAndGetFormData()
+            : RightPaneStore.getFormData().courseNumber;
     }
 
     state = {
         courseNumber: this.getCourseNumber(),
     };
 
-    handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-        this.setState({ courseNumber: event.target.value });
-        RightPaneStore.updateFormValue('courseNumber', event.target.value);
-        const url = new URL(window.location.href);
-        const urlParam = new URLSearchParams(url.search);
-        urlParam.delete('courseNumber');
-        if (event.target.value) {
-            urlParam.append('courseNumber', event.target.value);
+    handleChange = (name: string) => (event: ChangeEvent<HTMLInputElement>) => {
+        if (name === 'upper') {
+            if (event.target.checked) {
+                this.setState({ courseNumber: '100-199' });
+                RightPaneStore.updateFormValue('courseNumber', '100-199');
+                const url = new URL(window.location.href);
+                const urlParam = new URLSearchParams(url.search);
+                urlParam.delete('courseNumber');
+
+                urlParam.append('courseNumber', '100-199');
+                const param = urlParam.toString();
```

Comment:
Seems like this is duplicate code between the two parts of this conditional statement. Can you move the stuff that's repeated outside the if/else?